### PR TITLE
Fix updater with rubygems

### PR DESCRIPTION
**Pull request by hsbt**
Fixes https://github.com/rubygems/rubygems/issues/2123

### CODE REVIEW

File: lib/rubygems/commands/setup_command.rb [hunk 1]
```diff
@@ -368,9 +370,16 @@ def install_default_bundler_gem
 
     bundler_spec = Gem::Specification.load(default_spec_path)
 
+    # Remove gemspec that was same version of vendoered bundler.
```
Comment (segiddins) [inline lib/rubygems/commands/setup_command.rb]: `vendored`

Comment (segiddins) [reply lib/rubygems/commands/setup_command.rb]: also, what is the motivation for removing the user-installed Bundler gem spec?

File: lib/rubygems/commands/setup_command.rb [hunk 2]
```diff
@@ -368,9 +370,16 @@ def install_default_bundler_gem
...
+    normal_gemspec = File.join(Gem.default_dir, "specifications", "#{bundler_spec.version}.gemspec")
+    if File.file? normal_gemspec
+      File.delete normal_gemspec
+    end
+
+    # Remove gem files that were same version of vendored bundler.
     if File.directory? bundler_spec.gems_dir
       Dir.entries(bundler_spec.gems_dir).
-        select {|default_gem| File.basename(default_gem).match(/^bundler-#{Gem::Version::VERSION_PATTERN}$/) }.
+        select {|default_gem| File.basename(default_gem).match(/^bundler-#{bundler_spec.version}$/) }.
```
Comment (segiddins) [inline lib/rubygems/commands/setup_command.rb]: you'll want to escape the version, since it will contain regexp metacharacters

**Comment by hsbt** [comment]
@segiddins I added comments for a test scenario. rails/rails and other projects have following environment in Travis.

1. Travis install bundler-1.16.1
2. User invoke `gem update --system`
3. User invoke `gem install bundler`

Therefore, `update_rubygems` removed bundler-1.16.1 files and install vendored bundler-1.16.0. When a user invokes `bundle` command, rubygems activate 1.16.1 executable. So, User faced https://travis-ci.org/rails/rails/builds/319911098 .

We should not remove a `bundler-*` files. We should promote only vendored bundler and remove its version files. (In master branch, it version is 1.16.1)

**Comment by hsbt** [comment]
@bundlerbot r+

**Comment by segiddins** [comment]
But what happens when the user updates to a newer version of the vendored Bundler? Do we still remove the gemfile for the old version?

**Comment by hsbt** [comment]
>Do we still remove the gemfile for the old version?

What mean "the gemfile"?

**Comment by segiddins** [comment]
I meant the gemspec, sorry

**Comment by hsbt** [comment]
I see. I reconsider it again next week.

**Pull request merged**